test/e2e/volume_provisioning.go

```diff
@@ -108,61 +216,182 @@ var _ = framework.KubeDescribe("Dynamic Provisioning", func() {
+				"kubernetes.io/aws-ebs",
+				map[string]string{
+					"type": "gp2",
+					// Check that AWS can parse "zone" parameter, however
+					// we can't create PDs in different than default zone
+					// as we don't know zone names
+					"zone": framework.TestContext.CloudConfig.Zone,
+				},
+				"1.5Gi",
+				"2Gi",
+				func(volume *v1.PersistentVolume) error {
+					return checkAWSEBS(volume, "gp2", false)
+				},
+			},
+			{
+				"should provision io1 EBS on AWS",
+				[]string{"aws"},
+				"kubernetes.io/aws-ebs",
+				map[string]string{
+					"type":      "io1",
+					"iopsPerGB": "50",
+				},
+				"3.5Gi",
+				"4Gi", // 4 GiB is minimum for io1
+				func(volume *v1.PersistentVolume) error {
+					return checkAWSEBS(volume, "io1", false)
+				},
+			},
+			{
+				"should provision sc1 EBS on AWS",
+				[]string{"aws"},
+				"kubernetes.io/aws-ebs",
+				map[string]string{
+					"type": "sc1",
+				},
+				"500Gi", // minimum for sc1
+				"500Gi",
+				func(volume *v1.PersistentVolume) error {
+					return checkAWSEBS(volume, "sc1", false)
+				},
+			},
+			{
+				"should provision st1 EBS on AWS",
+				[]string{"aws"},
+				"kubernetes.io/aws-ebs",
+				map[string]string{
+					"type": "st1",
+				},
+				"500Gi", // minimum for st1
+				"500Gi",
+				func(volume *v1.PersistentVolume) error {
+					return checkAWSEBS(volume, "st1", false)
+				},
+			},
+			{
+				"should provision encrypted EBS on AWS",
+				[]string{"aws"},
+				"kubernetes.io/aws-ebs",
+				map[string]string{
+					"encrypted": "true",
+				},
+				"1Gi",
+				"1Gi",
+				func(volume *v1.PersistentVolume) error {
+					return checkAWSEBS(volume, "gp2", true)
+				},
+			},
+			// OpenStack generic tests (works on all OpenStack deployments)
+			{
+				"should provision generic Cinder volume on OpenStack",
+				[]string{"openstack"},
+				"kubernetes.io/cinder",
+				map[string]string{},
+				"1.5Gi",
+				"2Gi",
+				nil, // there is currently nothing to check on OpenStack
+			},
+			{
+				"should provision Cinder volume with empty volume type and zone on OpenStack",
+				[]string{"openstack"},
+				"kubernetes.io/cinder",
+				map[string]string{
+					"type":         "",
+					"availability": "",
+				},
+				"1.5Gi",
+				"2Gi",
+				nil, // there is currently nothing to check on OpenStack
+			},
+			// vSphere generic test
+			{
+				"should provision generic vSphere",
+				[]string{"vsphere"},
+				"kubernetes.io/vsphere-volume",
+				map[string]string{},
+				"1.5Gi",
+				"1.5Gi",
+				nil,
+			},
+		}
 
-	framework.KubeDescribe("DynamicProvisioner Beta", func() {
-		It("should create and delete persistent volumes [Slow] [Volume]", func() {
-			framework.SkipUnlessProviderIs("openstack", "gce", "aws", "gke")
+		var betaTest *storageClassTest
+		for i, t := range tests {
+			// Beware of clojure, use local variables instead of those from
+			// outer scope
+			test := t
+			suffix := fmt.Sprintf("%d", i)
+			It(test.name+" [Slow] [Volume]", func() {
```

Comment:
each storage class gets its own name, see `suffix := fmt.Sprintf("%d", i)` a few lines below. And all of them are non-default.